### PR TITLE
Update ADAS to support the latest Cherab version

### PR DESCRIPTION
**Pull request by vsnever**
This PR updates `cherab.adas` to support the latest Cherab version. To work properly, this update requires some features introduced in the PR cherab/core#246. The ADAS module is rewritten using the `cherab.openadas` API. Unlike `cherab.openadas`, this module creates a repository using the ADF files from the local ADAS installation. The interface to ADAS405 (ionisation ballance) is updated and the new interface to ADAS603 (Zeeman spectral line structure) is added. The method `zeeman_structure()` available exclusively in `ADAS` is required to unleash the full potential of the `ZeemanMultiplet` line shape model.

The demos for calculating the fractional abundance and total radiation are updated. A new demo for calculating the Zeeman splitting is added.

An access to the full ADAS package is required to review this PR. @jacklovell and @Mateasek, can I ask you to review this, because you have an access to the full ADAS via the JET cluster?

@mattngc, could you please create a development branch, because it would be safer to merge this with development?

### CODE REVIEW

File: README.md
Comment (jacklovell) [inline README.md]: It needs more than this: one needs to source one of the adas setup scripts `/home/adas/setup/adas_setup.csh` (tcsh) or `/home/adas/setup/adas_setup.ksh` (sh/bash/ksh). Sourcing those scripts is enough to set up the Python library, but adds other required environment variables such as `ADASFORT` which are required for the ADAS routines to work.

File: demos/total_radiation.py
Comment (jacklovell) [inline demos/total_radiation.py]: This can be specified as a scalar: the `sample2d_grid` function automatically converts scalars into length-1 arrays. A scalar here would more explicitly show that you're only sampling for a single density.

File: demos/total_radiation.py
Comment (jacklovell) [inline demos/total_radiation.py]: Try to avoid hard-coding `11` both times here: it's easy to update one and forget the other at a later time. Maybe `for charge in range(fractional_abundance.shape[0])`. Or just set a variable `num_charges = 11` and use that in the `np.zeros` and `range` calls.

You could also do the following, since iterating over a 2D numpy array returns views of the rows, but it may be overcomplicating things for a simple demo.
```
for charge, abundance in enumerate(fractional_abundance):
    fraction = atomic_data.fractional_abundance(neon, charge)
    abundance[:] = sample2d_grid(fraction, electron_density, electron_temperature).squeeze()
```

File: demos/total_radiation.py
Comment (jacklovell) [inline demos/total_radiation.py]: Same comment about avoiding hard-coding the value 10.

File: cherab/adas/repository/atomic.py
Comment (jacklovell) [inline cherab/adas/repository/atomic.py]: I don't think we should monkey-patch modules in `cherab.openadas` with the ADAS repository path. This could cause hard-to-find bugs if users somehow tried or wanted to use OpenADAS and ADAS data in the same program. All the functions in `cherab.openadas.repository` accept a `repository_path` argument: I think a more robust way to deal with this is to simply wrap all the functions in the `cherab.openadas.repository` module with `repository_path=cherab.adas.repository.utility.DEFAULT_REPOSITORY_PATH`. For this, [functools.wraps](https://docs.python.org/3/library/functools.html#functools.wraps) will be useful.

File: cherab/adas/repository/beam/cx.py
Comment (jacklovell) [inline cherab/adas/repository/beam/cx.py]: Same comment re. monkey-patching.

File: cherab/adas/repository/beam/emission.py
Comment (jacklovell) [inline cherab/adas/repository/beam/emission.py]: Same comment re. monkey-patching.

File: cherab/adas/repository/beam/population.py
Comment (jacklovell) [inline cherab/adas/repository/beam/population.py]: Same comment re. monkey-patching.

File: cherab/adas/repository/beam/stopping.py
Comment (jacklovell) [inline cherab/adas/repository/beam/stopping.py]: Same comment re. monkey-patching.

File: cherab/adas/repository/pec.py
Comment (jacklovell) [inline cherab/adas/repository/pec.py]: Same comment re. monkey-patching.

File: cherab/adas/repository/radiated_power.py
Comment (jacklovell) [inline cherab/adas/repository/radiated_power.py:31]: Same comment re. monkey-patching.

File: cherab/adas/repository/wavelength.py
Comment (jacklovell) [inline cherab/adas/repository/wavelength.py]: Same comment re. monkey-patching.

**Review by jacklovell — CHANGES_REQUESTED** [review]
Thanks Vlad. I think this is a very useful change to make. I've requested a few changes, but on the whole it looks good.

I think we should hold off on merging it, even into a development branch in this repository, until #246 in core is merged into development.

File: README.md
Comment (vsnever) [reply README.md]: Thanks, I updated the instruction:

Set the ADAS environment variables before using this package by running source `/home/adas/setup/adas_setup.ksh` if your Unix shell is sh, bash or ksh, or source `/home/adas/setup/adas_setup.csh` if your Unix shell is tcsh.

File: demos/total_radiation.py
Comment (vsnever) [reply demos/total_radiation.py]: Done.

File: demos/total_radiation.py
Comment (vsnever) [reply demos/total_radiation.py]: Done.

File: demos/total_radiation.py
Comment (vsnever) [reply demos/total_radiation.py]: Done.

File: cherab/adas/repository/atomic.py
Comment (vsnever) [reply cherab/adas/repository/atomic.py]: Yes, I've used the laziest possible solution. Your solution is better. Now all the functions from `cherab.openadas.repository` are wrapped in `repository_path_decorator`. Please, let me know if this wrapper can be improved.

**Comment by vsnever** [comment]
Thank you very much for the review, @jacklovell. I tried to improve the code as you suggested. I agree that we should wait until  cherab/core#246 is approved and merged.

**Review by jacklovell — APPROVED** [review]
Thanks for the additional updates. Looks good to me now.

**Pull request merged**
